### PR TITLE
Remove f16c flag in utils_sse

### PR DESCRIPTION
**Pull request by cqy123456**
issue: #815 
Milvus issue: https://github.com/milvus-io/milvus/issues/35877

### CODE REVIEW

File: src/simd/hook.cc [hunk 1]
```diff
@@ -89,12 +89,6 @@ cpu_support_sse4_2() {
     InstructionSet& instruction_set_inst = InstructionSet::GetInstance();
     return (instruction_set_inst.SSE42());
 }
-
```
Comment (alexanderguzhva) [inline src/simd/hook.cc:92]: I would keep this function just in case

File: src/simd/distances_sse.cc [hunk 2]
```diff
@@ -67,20 +68,12 @@ fvec_norm_L2sqr_sse(const float* x, size_t d) {
 
 float
 fp16_vec_norm_L2sqr_sse(const knowhere::fp16* x, size_t d) {
-    __m128 m_res = _mm_setzero_ps();
-    while (d >= 4) {
-        __m128 m_x = _mm_cvtph_ps(_mm_loadl_epi64((const __m128i_u*)x));
-        m_res = _mm_add_ps(m_res, _mm_mul_ps(m_x, m_x));
-        x += 4;
-        d -= 4;
+    std::vector<knowhere::fp32> fp32_x(d);
```
Comment (alexanderguzhva) [inline src/simd/distances_sse.cc:71]: `std::vector` introduces an enormous overhead.
Please implement these functions manually in the following way:
```
    float res = 0;
    for (size_t i = 0; i < d; i++) {
        const float xv = float(x[i]);
        res += xv * xv;
    }

    return res;
``` 
Similar for dot products

File: src/simd/hook.h [hunk 3]
```diff
@@ -103,8 +103,6 @@ bool
 cpu_support_avx2();
 bool
 cpu_support_sse4_2();
-bool
```
Comment (alexanderguzhva) [inline src/simd/hook.h:106]: I'd keep this function

**Review by alexanderguzhva — CHANGES_REQUESTED** [review]

**Comment by alexanderguzhva** [comment]
@cqy123456 We can use `_ref` versions, sure. But the downside is that a compiler will not use `sse` instructions for compiling `ref` versions, such a change introduces a performance regression. Therefore, the most optimal way is to literally copy-paste `ref` versions code into `distances_sse.cc` under new function names, because it allows a compiler to use `sse` instructions. 
It is not a big problem if we're going to have small pieces of the same code in different files.

**Comment by cqy123456** [comment]
At first I tried to only rely on compiler optimization, but the compiler did not speed up fp16's (float)x * (float)y very well, probably because the fp16->float operation is more complicated.

**Comment by PwzXxm** [comment]
/lgtm
Patch to release for correctness first.

**Pull request merged**